crates/generate_artifacts/src/link_type.rs

```diff
@@ -0,0 +1,28 @@
+use common_lang_types::{
+    ArtifactPathAndContent, ParentObjectEntityNameAndSelectableName, ServerObjectEntityName,
+};
+use intern::string_key::Intern;
+
+use crate::generate_artifacts::LINK_TYPE_FILE_NAME;
+
+pub fn generate_link_type_artifact<TNetworkProtocol: isograph_schema::NetworkProtocol>(
+    schema: &isograph_schema::Schema<TNetworkProtocol>,
+    server_object_entity_name: &ServerObjectEntityName,
+) -> ArtifactPathAndContent {
+    let link_type = TNetworkProtocol::generate_link_type(schema, server_object_entity_name);
+
+    let link_type_content = format!(
+        "import type {{ Link }} from '@isograph/react';\n
+export type {server_object_entity_name}LinkFutureType = Link<\"%{server_object_entity_name} future type%\">;\n
```

Comment:
I like that this is generated anew for each type, that seems better than having a single branded type for all future types 👍 

I am a bit torn, though, because it is nice to have a single "future type" to check against, which makes review etc easier.

If we have a single future type, let's export it from @isograph/react

I don't feel very strongly either way